cachito/web/models.py

```diff
@@ -179,15 +180,46 @@ def to_purl(self):
 
         :return: the PURL string of the Package object
         :rtype: str
-        :raise ValueError: if the there is no implementation for the package type
+        :raise ContentManifestError: if the there is no implementation for the package type
         """
         if self.type in ("go-package", "gomod"):
             # Use only the PURL "name" field to avoid ambiguity for Go modules/packages
             # see https://github.com/package-url/purl-spec/issues/63 for further reference
             purl_name = self.name.replace("/", "%2F")
             return f"pkg:golang/{purl_name}@{self.version}"
+
+        elif self.type == "npm":
+            purl_name = self.name.replace("@", "%40")
```

Comment:
So the [documentation](https://docs.npmjs.com/files/package.json#name) says "the name ends up being part of a URL, an argument on the command line, and a folder name. Therefore, the name can’t contain any non-URL-safe characters.".

Keep in mind though, the user could put whatever they want in that field and Cachito would honor it since just reads the `package.json` file. It might be best to just URL encode to be safe.

Reply:
I encoded it (and the go one as well) to account for any unexpected special characters.